src/main/java/org/junitpioneer/jupiter/ReportEntryExtension.java

```diff
@@ -33,4 +42,31 @@ private static void verifyKeyValueAreNotBlank(ReportEntry entry) {
 		}
 	}
 
+	@Override
+	public void testDisabled(ExtensionContext context, Optional<String> reason) {
+		// If the test is disabled we consider the annotation disabled too and don't publish anything.
+	}
+
+	@Override
+	public void testSuccessful(ExtensionContext context) {
+		publishOnConditions(context, ALWAYS, ON_SUCCESS);
+	}
+
+	@Override
+	public void testAborted(ExtensionContext context, Throwable cause) {
+		publishOnConditions(context, ALWAYS, ON_FAILURE);
+	}
+
+	@Override
+	public void testFailed(ExtensionContext context, Throwable cause) {
+		publishOnConditions(context, ALWAYS, ON_FAILURE);
+	}
+
+	private void publishOnConditions(ExtensionContext context, ReportEntry.PublishCondition... conditions) {
+		findAnnotations(context)
+				.filter(entry -> Arrays.asList(conditions).contains(entry.when()))
+				.filter(entry -> !entry.key().isEmpty() && !entry.value().isEmpty())
```

Comment:
Is this line necessary as non-empty keys and values are checked in line 31 ?

Reply:
It's necessary, if you want to avoid throwing an exception from TestReporter.

Reply:
I don't get it either. As far as I understand the code, this line will only ever do something if `verifyKeyValueAreNotBlank` is broken by a change. But such a change would fail a test and thus not go undetected. I hence don't see a scenario where this line changes anything.

Reply:
We throw an `ExtensionConfigurationException` before test execution if either key or value are blank. This leads to the test failing. So, JUnit calls the `TestWatcher::testFailed`. If the publish condition is `ALWAYS` or `ON_FAILURE`, we try to publish report entries. So this is filtering out *why* the test failed - we don't want to publish if the extension configuration is incorrect.

Reply:
Added a short comment that (tries to) concisely explains this.